packages/loopring_v3/contracts/aux/access/LoopringIOExchangeOwner.sol

```diff
@@ -32,7 +33,14 @@ contract LoopringIOExchangeOwner is SelectorBasedAccessManager
             hasAccessTo(msg.sender, SUBMITBLOCKS_SELECTOR) || open,
             "PERMISSION_DENIED"
         );
+        uint _gasleft = gasleft();
         bytes memory decompressed = LzDecompressor.decompress(data);
+
+        // Log the gas reduction in calldata and the cost in data decompression.
+        // If (decompressed.length - data.length) * 16 > (_gasleft - gasleft()), then
```

Comment:
Not really accurate like this because non-zero bytes will cost 16 gas, but zero bytes will cost 4 gas. And the compression will mostly target the zero bytes.

I don't think we'll be able to measure this well on-chain, I think we'll have to do the measurements on the actual Ethereum tx, there are also hidden costs elsewhere that can easily be overlooked.